lib/mm_code.rb

```diff
@@ -0,0 +1,76 @@
+CONVERSION_CONTAINER = {
+    1 => 'R',
+    2 => 'G',
+    3 => 'Y',
+    4 => 'B',
+    5 => 'M',
+    6 => 'C'
+}
+
+COLOR_CONTAINER = {
+    1 => "\e[91m●\e[0m",
+    2 => "\e[92m●\e[0m",
+    3 => "\e[93m●\e[0m",
+    4 => "\e[94m●\e[0m",
+    5 => "\e[95m●\e[0m",
+    6 => "\e[96m●\e[0m"
+}
+
+class Code
+    attr_reader :value, :matched
+
+    def initialize(new_code)
+        if /[rgybmcRGYBMC]/ === new_code[0]
+            new_code = new_code.join.upcase.split(//)
+            @value = convert(new_code)
+        else
+            @value = new_code
+        end
+    end
```

Comment:
Not a great pattern to have branching behavior in a constructor. Is there a way around this?